### DISCUSSION TITLE
Send a query back into a loading state (Suspense)

### DISCUSSION THREAD

**Discussion by iway1** [TanStack/query#3626] [General] [answered]
As far as I can tell, there doesn't seem to be a way to get a fetched query to go back into a loading state and for it to suspend the component again after it's being fetched? Is there a mechanism for this? 

I have a use case where I would like my data to usually not show a loading state when background fetches are happening, but there is a certain circumstance where I do want to show it in a loading state while refetching? Any general solution for this without me having to create my own state management? If I can put the query back into loading state it would solve this for me I think

**Comment by TkDodo** [comment, marked as answer]
you can remove the query with `queryClient.removeQueries` or with `remove` returned from `useQuery`. it will remove the value from the cache, but not re-render any subscribers. If the component re-renders again, it will go into hard loading state, because it has no data, thus re-throwing to the suspense boundary.

**Reply by iway1** [reply]
Ah okay that's very informative. So in my case I would need to trigger a rerender for a subscriber - it sounds like that's something I'd need to do outside of react-query?

**Reply by TkDodo** [reply]
if you want to remove + re-render, `queryClient.resetQueries` is your friend because it does exactly that.

**Reply by iway1** [reply]
U r the man! Thanks!
